api/types.go

```diff
@@ -74,6 +74,9 @@ type GenerateRequest struct {
 	// request, for multimodal models.
 	Images []ImageData `json:"images,omitempty"`
 
+	// Tools is an optional list of tools the model has access to.
+	Tools []Tool `json:"tools,omitempty"`
```

Comment:
Does this make sense to have in `/api/generate` at all?

Reply:
I don't see why not. The template ends up being the same

Reply:
We may want to leave it out (for now) given that the `tool` role is required to send back the tool responses